src/pymor/algorithms/basic.py

```diff
@@ -34,38 +33,27 @@ def almost_equal(U, V, product=None, norm=None, rtol=1e-14, atol=1e-14):
         |VectorArrays| to be compared.
     product
         If specified, use this inner product |Operator| to compute the norm.
-        `product` and `norm` are mutually exclusive.
-    norm
-        If specified, must be a callable which is used to compute the norm
-        or, alternatively, one of the strings 'l1', 'l2', 'sup', in which case the
-        respective |VectorArray| norm methods are used.
-        `product` and `norm` are mutually exclusive. If neither is specified,
-        `norm='l2'` is assumed.
+        Otherwise, the Euclidean norm is used.
+    sup_norm
+        If specified, use the sup norm to compute the error.
     rtol
         The relative tolerance.
     atol
         The absolute tolerance.
     """
 
-    assert product is None or norm is None
-    assert not isinstance(norm, str) or norm in ('l1', 'l2', 'sup')
-    norm = induced_norm(product) if product is not None else norm
-    if norm is None:
-        norm = 'l2'
-    if isinstance(norm, str):
-        norm_str = norm
-        norm = lambda U: getattr(U, norm_str + '_norm')()
+    assert product is None or not sup_norm 
```

Comment:
Shouldn't this now be
`assert sup_norm or product is not None` ?

Reply:
No, I think @sdrave's version is correct.

`assert product is None or not sup_norm` is the same as `assert not (product is not None and sup_norm)` and I think that is what one wants to check here. You do not want that a user provides a product and at the same time sets `sup_norm` to True.

Reply:
Alright, If that's the intention then I misread the docstring change.

Reply:
I think the docstring is not optimal.

```
sup_norm
        If specified, use the sup norm to compute the error.
```
sounds like `sup_norm` gets some `norm` as value. Maybe something like
```
sup_norm
        If `True`, use the sup norm to compute the error.
```
fits better.

Reply:
Fixed the docstring.